contribs/gmf/src/services/themesservice.js

```diff
@@ -103,24 +120,51 @@ gmf.Themes.findTheme_ = function(themes, themeName) {
  * @return {angular.$q.Promise} Promise.
  */
 gmf.Themes.prototype.getBgLayers = function() {
+
   goog.asserts.assert(!goog.isNull(this.promise_));
   return this.promise_.then(goog.bind(
       /**
        * @param {gmf.ThemesResponse} data The "themes" web service response.
-       * @return {Array.<Object>} Array of background layer objects.
+       * @return {angular.$q.Promise}
        */
       function(data) {
-        var bgLayers = data['background_layers'].map(goog.bind(function(item) {
+        var configs = data['background_layers'];
+        this.backgroundLayerConfigs_ = configs;
+        var promises = configs.map(goog.bind(function(item) {
+          return this.$http_.get(item.url, {
+            cache: false
+          });
+        }, this));
+        return this.q_.all(promises);
```

Comment:
`$q.all` could be a good approach, but i'm not sure that the callback function of your promise will have the array of background layers as its argument.